lib/driver.js

```diff
@@ -399,6 +399,18 @@ class XCUITestDriver extends BaseDriver {
       }
     }
 
+    if (this.opts.permissions) {
+      if (this.isSimulator()) {
+        log.debug('Setting the requested permissions before WDA is started');
+        for (const [bundleId, permissionsMapping] of JSON.parse(this.opts.permissions)) {
```

Comment:
Should there be validation that the `permissions` is valid JSON? This will fail pretty unclearly if it is not (along the lines of `SyntaxError: Unexpected token h in JSON at position 1`).

I would expect something in `validateDesiredCaps`.

Reply:
I'm confused. This verification IS there

Reply:
Sorry. I missed that. Github has changed its diff format and it's not longer clear what function things are in.